README.md

```diff
@@ -0,0 +1,432 @@
+# OLab45-Api
+
+OLab 4.5 backend REST API
+
+## Install
+
+The API service requires .NET 6+ and MySQL or MariaDB installed. The instructions below are provided for RHEL and RHEL-like systems such as CentOS Stream.
+
+### Prerequisites
+
+The following commands will install the mysql server/client, nginx, in addition to some packages needed for the build, and whitelist the http/https ports for `firewalld`.
+
+```sh
+# update packages
+sudo dnf update
+
+# install mariadb server and client - https://mariadb.com/kb/en/yum/
+sudo dnf install mariadb-server
+
+# enable and start mariadb service
+sudo systemctl enable mariadb
+sudo systemctl start mariadb
+
+# inital mysql server setup
+sudo mysql_secure_installation
+
+# install nginx
+sudo dnf install nginx
+
+# enable and start nginx service
+sudo systemctl enable nginx
+sudo systemctl start nginx
+
+# enable http and https if using firewalld
+sudo firewall-cmd --permanent --add-service=http --add-service=https
+sudo firewall-cmd --reload
+
+# install extra dependencies needed
+sudo dnf install git wget tar
+```
+
+### .NET 6
+
+As a root user, download the installer and install dotnet sdk from the `6.0` channel:
+
+```sh
+# get install helper
+wget https://dot.net/v1/dotnet-install.sh -O /usr/bin/dotnet-install
```

Comment:
Isn't .net 6 a native package on RHEL as it is on Ubuntu?  Installing it via the native packet manager against the microsoft repo makes it way easier to keep track of updates.

Reply:
For RHEL, that's correct. I was testing on a CentOS 8 VM yesterday when I made the update. Prior to this, that package can be installed from the official RPM `dnf install dotnet-sdk-6.0`.

I was able to build the APIs prior to your recent `dev -> main` merge using the official install, but yesterday it was no longer the case on CentOS.

I was unable to test on the RHEL VM provided since it has disc space issues. I'll restore to `sudo dnf install dotnet-sdk-6.0` while trusting this will install a non-pre-release version on RHEL, it was in fact installing an RC on CentOS Stream and since the app requires `.NET 6.0.0`, the RC tag will decrease the version minor making the runtime/sdk incompatible ([more info](https://github.com/dotnet/runtime/issues/64862#issuecomment-1030860252)).